feat: add devfile schema validation to the generator

svor opened the pull request:
<!-- Please review the following before submitting a PR:
Che's Contributing Guide: https://github.com/eclipse/che/blob/master/CONTRIBUTING.md
Pull Request Policy: https://github.com/eclipse/che/wiki/Development-Workflow#pull-requests
-->

### What does this PR do?
Integrates devfile schema validator into the devworkspace generator tool. The validator checks devfile before generating a dev workspace. In case if the devfile is not valid, it'll show validation problems.

The validator takes into the account next devfile's schemes: **2.0.0**, **2.1.0**, **2.2.0** and  **2.2.1-alpha**

### Screenshot/screencast of this PR
<!-- Please include a screenshot or a screencast explaining what is doing this PR -->

![screenshot-nimbusweb me-2023 11 09-13_10_19](https://github.com/eclipse-che/che-devfile-registry/assets/1271546/dd9772e2-6818-4e80-94c0-01bccc54dea1)


### What issues does this PR fix or reference?
<!-- Please include any related issue from eclipse che repository (or from another issue tracker).
     Include link to other pull requests like documentation PR from [the docs repo](https://github.com/eclipse/che-docs)
-->
https://github.com/eclipse/che/issues/22594

### How to test this PR?
<!-- Please explain for example :
  - The test platform (openshift, kubernetes, minikube, CodeReady Container, docker-desktop, etc)
  - Installation method: chectl / che-operator
  - steps to reproduce
 -->
1. By using a command line:
```bash
npx @svor/che-devworkspace-generator@7.77.0-7-next \
        --devfile-url:https://github.com/svor/web-nodejs-sample/tree/no-valid-devfile \
        --plugin-registry-url:https://che-plugin-registry-main.surge.sh/v3/ \
        --editor-entry:che-incubator/che-code/latest \
        --output-file:/tmp/devworkspace-che-code-latest.yaml
```
The generator should fail with a message:
```
Validating devfile
stack=Error: Devfile schema validation failed. Error: 0: instance.commands[0].exec requires property "commandLine"
1: instance.commands[0].exec requires property "component"
```

2. Uing the User dashboard. 

For this need to deploy Che with custom dashboard image: quay.io/vsvydenk/che-dashboard:new-generator. This image uses a new version of the devworkspace generator and provides a new sample with invalid devfile.

To test changes just try to create a workspace from **Empty Workspace Invalid Command** sample. This sample uses invalid devfile:
```yaml
schemaVersion: 2.2.0
metadata:
  name: empty
  namespace: svor-che
components:
  - name: universal-developer-image
    container:
      image: quay.io/devfile/universal-developer-image:ubi8-latest
commands:
  - id: command-id
    exec:
      label: "2. Configure Che to use the Custom Devfile Registry"
      # invalid devfile: no componnet
      # component: builder
      # invalid devfile: no commandLine
      # commandLine: ls
``` 
As a result you should see 

![screenshot-nimbusweb me-2023 11 09-13_10_19](https://github.com/eclipse-che/che-devfile-registry/assets/1271546/ad374d92-3229-4aa6-8991-0ce64188c521)


### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)

- [x] [The Eclipse Contributor Agreement is valid](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-eclipse-contributor-agreement-is-valid)
- [x] [Code produced is complete](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-produced-is-complete)
- [x] [Code builds without errors](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-builds-without-errors)
- [x] [Tests are covering the bugfix](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#tests-are-covering-the-bugfix)
- [x] [The repository devfile is up to date and works](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-repository-devfile-is-up-to-date-and-works)
- [x] [Sections `What issues does this PR fix or reference` and `How to test this PR` completed](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#sections-what-issues-does-this-pr-fix-or-reference-and-how-to-test-this-pr-completed)
- [x] [Relevant user documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [x] [Relevant contributing documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [x] [CI/CD changes implemented, documented and communicated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#cicd-changes-implemented-documented-and-communicated)

### Reviewers

Reviewers, please comment how you tested the PR when approving it.

ibuziuk reviewed (commented):
@svor 2.2.2 has been released recently so we should add it to the validation and probably remove 2.2.1 alpha from this PR

svor commented on tools/devworkspace-generator/package.json:
```diff
@@ -41,6 +41,7 @@
   "homepage": "https://github.com/eclipse-che/che-devfile-registry#readme",
   "dependencies": {
     "@devfile/api": "2.2.1-alpha-1667236163",
```
the version of devfile/api library will be updated as a result of https://github.com/eclipse/che/issues/22658

svor commented:
> @svor 2.2.2 has been released recently so we should add it to the validation and probably remove 2.2.1 alpha from this PR

- added schema validation for 2.2.1 and 2.2.2
- removed schema validation for 2.2.1_alpha

olexii4 reviewed (approved):
Awesome! all works as expected.
